packages/atlas/src/hooks/useLogIn.ts

```diff
@@ -0,0 +1,112 @@
+import { JOYSTREAM_ADDRESS_PREFIX } from '@joystream/types'
+import { ScryptOpts, scrypt } from '@noble/hashes/scrypt'
+import { Keyring } from '@polkadot/keyring'
+import { u8aToHex } from '@polkadot/util'
+import { cryptoWaitReady } from '@polkadot/util-crypto'
+import axios from 'axios'
+import { entropyToMnemonic } from 'bip39'
+import { Buffer } from 'buffer'
+import { AES, enc, lib, mode } from 'crypto-js'
+import { useCallback } from 'react'
+
+import { ORION_AUTH_URL } from '@/config/env'
+import { SentryLogger } from '@/utils/logs'
+
+export const keyring = new Keyring({ type: 'sr25519', ss58Format: JOYSTREAM_ADDRESS_PREFIX })
+
+// todo extract these 3 Fn to `packages/atlas/src/utils/user.ts` after #4168 is merged
+export async function scryptHash(
+  data: string,
+  salt: Buffer | string,
+  options: ScryptOpts = { N: 32768, r: 8, p: 1, dkLen: 32 }
+): Promise<Buffer> {
+  return new Promise((resolve) => {
+    resolve(Buffer.from(scrypt(Buffer.from(data), salt, options)))
+  })
+}
+
+function aes256CbcDecrypt(encryptedData: string, key: Buffer, iv: Buffer): string {
+  const keyWA = enc.Hex.parse(key.toString('hex'))
+  const ivWA = enc.Hex.parse(iv.toString('hex'))
+  const decrypted = AES.decrypt(lib.CipherParams.create({ ciphertext: enc.Hex.parse(encryptedData) }), keyWA, {
+    iv: ivWA,
+    mode: mode.CBC,
+  })
+  return decrypted.toString(enc.Hex)
+}
+
+const getArtifacts = async (id: string) => {
+  try {
+    const res = await axios.get<{ cipherIv: string; encryptedSeed: string }>(`${ORION_AUTH_URL}/artifacts?id=${id}`)
+
+    return res.data
+  } catch (error) {
+    SentryLogger.error('Error when fetching artifacts', 'useLogIn', error)
+  }
+}
+
+export enum LogInErrors {
+  ArtifactsNotFound = 'ArtifactsNotFound',
+  LoginError = 'LoginError',
+}
+
+type LogInHandler = {
+  data: {
+    accountId: string
+  } | null
+  error?: LogInErrors
+}
+
+export const useLogIn = () => {
```

Comment:
I don't think that we need to create a hook, the function doesn't use any state or other "react toys", this could be just `logIn`.

Reply:
I think we might need to set some global state after logging in, so I will keep it till it clears out